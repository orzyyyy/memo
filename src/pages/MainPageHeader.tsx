import React, { useState } from 'react';
import { Button, Input, Select } from 'antd';
import { MainPageProps } from './MainPage';
import { Plus } from '@ant-design/icons';
import { SelectValue } from 'antd/lib/select';
const { Option } = Select;

const MainPageHeader = ({
  Header,
  onExhentaiDownload,
  onEdit,
  exhentaiDateSet,
  onExhentaiSelectChange,
}: MainPageProps & { Header: any }) => {
  const [selectValue, setSelectValue] = useState();

  const handleSelectChange = (value: SelectValue) => {
    onExhentaiSelectChange(value);
    setSelectValue(value as any);
  };

  return (
    <Header
      style={{
        background: 'rgba(0, 0, 0, 0)',
        height: 48,
        position: 'relative',
      }}
    >
      <Select
        style={{ position: 'absolute', width: 160, top: 10, left: 8 }}
        value={
          selectValue || (exhentaiDateSet.length ? exhentaiDateSet[0] : '')
        }
        onChange={handleSelectChange}
      >
        {exhentaiDateSet.map(timeStamp => (
          <Option value={timeStamp} key={`exhentai-time-stamp-${timeStamp}`}>
            {timeStamp}
          </Option>
        ))}
      </Select>
      <Input
        onPressEnter={onExhentaiDownload}
        style={{ position: 'absolute', right: 80, top: 10, width: 350 }}
      />
      <Button
        style={{ position: 'absolute', right: 24, top: 10 }}
        onClick={() => onEdit()}
      >
        <Plus />
      </Button>
    </Header>
  );
};

export default MainPageHeader;
