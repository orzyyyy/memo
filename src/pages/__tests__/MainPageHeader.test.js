import React from 'react';
import { shallow, mount } from 'enzyme';
import { Layout } from 'antd';
const Header = Layout.Header;
import MainPageHeader from '../MainPageHeader';

describe('MainPageHeader', () => {
  const errorSpy = jest.spyOn(console, 'error').mockImplementation(() => {});

  afterAll(() => {
    errorSpy.mockRestore();
  });

  it('render correctly', () => {
    const wrapper = shallow(
      <MainPageHeader Header={Header} exhentaiDateSet={[]} />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('onEdit of init button works correctly', () => {
    const onEdit = jest.fn();
    const wrapper = mount(
      <MainPageHeader onEdit={onEdit} Header={Header} exhentaiDateSet={[]} />,
    );
    wrapper
      .find('Button')
      .at(1)
      .props()
      .onClick();
    expect(onEdit).toHaveBeenCalled();
  });
});
