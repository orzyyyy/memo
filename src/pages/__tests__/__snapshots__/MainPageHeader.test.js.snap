// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPageHeader render correctly 1`] = `
<Adapter
  style={
    Object {
      "background": "rgba(0, 0, 0, 0)",
      "height": 48,
      "position": "relative",
    }
  }
>
  <Select
    choiceTransitionName="zoom"
    onChange={[Function]}
    style={
      Object {
        "left": 8,
        "position": "absolute",
        "top": 10,
        "width": 160,
      }
    }
    transitionName="slide-up"
    value=""
  />
  <Button
    block={false}
    ghost={false}
    htmlType="button"
    loading={false}
    style={
      Object {
        "left": 180,
        "position": "absolute",
        "top": 10,
      }
    }
    type="dashed"
  >
    列表
  </Button>
  <Input
    style={
      Object {
        "position": "absolute",
        "right": 80,
        "top": 10,
        "width": 350,
      }
    }
    type="text"
  />
  <Button
    block={false}
    ghost={false}
    htmlType="button"
    loading={false}
    onClick={[Function]}
    style={
      Object {
        "position": "absolute",
        "right": 24,
        "top": 10,
      }
    }
  >
    <Plus />
  </Button>
</Adapter>
`;
